pkgs/ffigen/test/code_generator_tests/expected_bindings/_expected_unions_bindings.dart

```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
```

Comment:
Shouldn't only the new files have 2024? We don't need to update existing headers.

Reply:
The problem is that the header for these goldens in defined in `code_generator_test.dart` and shared between all of them. So whenever we add a new test, the health check will complain about a newly added file with an old year in its header and we have to update all of them. Since this PR has skipped those checks I have reverted these changes and instead given the new tests a 2023 header as well.

Reply:
Maybe we should make it conditional later, but lets leave that for another PR and lets get this in!